docs/notebooks/api_user_guide/4_search.ipynb

```diff
@@ -2144,672 +2349,686 @@
+     ]
     }
    ],
    "source": [
-    "dag.guess_product_type(platform=\"LANDSAT OR SENTINEL1\")"
+    "products, estimated_total_number = dag.search(\n",
+    "    random_parameter=\"random_value\",\n",
+    "    **default_search_criteria\n",
+    ")"
    ]
   },
   {
    "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "returns all product types where the platform is either LANDSAT or SENTINEL1."
+    "The logs above show that the random kwarg passed to the [search()](../../api_reference/core.rst#eodag.api.core.EODataAccessGateway.search) method was indeed propagated to the request made to *PEPS*. It happens that *PEPS* accept unsupported parameters and still returned products, another provider might just have sent an error."
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 71,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [
-    {
-     "data": {
-      "text/plain": [
-       "['LANDSAT_C2L1',\n",
-       " 'LANDSAT_C2L2',\n",
-       " 'LANDSAT_C2L2ALB_BT',\n",
-       " 'LANDSAT_C2L2ALB_SR',\n",
-       " 'LANDSAT_C2L2ALB_ST',\n",
-       " 'LANDSAT_C2L2ALB_TA',\n",
-       " 'LANDSAT_C2L2_SR',\n",
-       " 'LANDSAT_C2L2_ST',\n",
-       " 'LANDSAT_ETM_C2L1',\n",
-       " 'LANDSAT_ETM_C2L2',\n",
-       " 'LANDSAT_TM_C2L1',\n",
-       " 'LANDSAT_TM_C2L2',\n",
-       " 'S1_SAR_GRD',\n",
-       " 'S1_SAR_OCN',\n",
-       " 'S1_SAR_RAW',\n",
-       " 'S1_SAR_SLC']"
-      ]
-     },
-     "execution_count": 71,
-     "metadata": {},
-     "output_type": "execute_result"
-    }
-   ],
    "source": [
-    "dag.guess_product_type(keywords=\"(LANDSAT AND collection2) OR SAR\")"
+    "## Guess a product type"
    ]
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "returns all product types which contain either the keywords LANDSAT and collection2 or the keyword SAR."
+    "`eodag` has an internal product type catalog which stores a number of metadata for each product type (see [this page](../../getting_started_guide/product_types.rst) for more information). It is actually possible to search for products by using some of these metadata. In that case, `eodag` will query its own catalog and use **the product type that best matches the query** among the available product types (i.e. all product types offered by providers where the necessary credentials are provided). The supported parameters are:\n",
+    "\n",
+    "* `free_text`: *search applied to all of the following fields*\n",
+    "* `instrument` (e.g. *MSI*)\n",
+    "* `platform` (e.g. *SENTINEL2*)\n",
+    "* `platformSerialIdentifier` (e.g. *S2A*)\n",
+    "* `processingLevel` (e.g. *L1*)\n",
+    "* `sensorType` (e.g. *OPTICAL*)\n",
+    "* `keywords` (e.g. *SENTINEL2 L1C SAFE*), which is case insensitive and ignores `-` or `_` characters\n",
+    "* `md5` (e.g. *424e09bc6a2fd295529f57231d036eed*)\n",
```

Comment:
`md5` must not be exposed and is only used internally to know if index rebuild is needed.
Rename it to `_md5` ?